sympy/ntheory/residue_ntheory.py

```diff
@@ -628,18 +628,17 @@ def is_nthpow_residue(a, n, m):
     .. [1] P. Hackman "Elementary Number Theory" (2009), page 76
 
     """
+    a = a % m
```

Comment:
I suppose a test `assert is_nthpow_mod(Dummy(even=True)+3, 3, 2) == True` could be added.